craft_parts/packages/normalize.py

```diff
@@ -0,0 +1,263 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2017-2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Definition and helpers for the repository base class."""
+
+import contextlib
+import fileinput
+import glob
+import itertools
+import logging
+import os
+import re
+import shutil
+import stat
+from pathlib import Path
+from typing import TYPE_CHECKING, Optional, Pattern, Union
+
+if TYPE_CHECKING:
+    from .base import RepositoryType
+
+
+logger = logging.getLogger(__name__)
+
+
+def normalize(unpack_dir: str, *, repository: "RepositoryType") -> None:
```

Comment:
This is what I mentioned above, `unpack_dir` should be pathlib.Path.

Reply:
Refactored the normalize module to use pathlib.